Update issue template (thanks jart@) and add env collection script.

aselle opened the pull request.

aselle committed:
Update issue template (thanks jart@) and add env collection script.

drpngx commented on ISSUE_TEMPLATE.md:
Could we use GIT_VERSION and VERSION instead?

drpngx commented on tools/tf_env_collect.sh:
```diff
@@ -0,0 +1,83 @@
+echo '== tensorflow import ============================================' >> $OUTPUT_FILE
+cat <<EOF > /tmp/check_tf.py
+import tensorflow as tf;
+print("tf.__version__ = %s" % tf.__version__)
```
same here

aselle replied:
ok

drpngx commented on tools/tf_env_collect.sh:
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
```
bash -u

aselle replied:
That does not seem to work. What is it supposed to do?

aselle commented on ISSUE_TEMPLATE.md:
ok

drpngx commented:
It generates an error if a variable is undefined. That's good for spotting
typos. You can set -u in the script.

The style guide mandates -eu, but -e which causes the script to end on any
non success command is too harsh in that case

On Apr 11, 2017 11:11 PM, "Andrew Selle" <notifications@github.com> wrote:

> *@aselle* commented on this pull request.
> ------------------------------
>
> In tools/tf_env_collect.sh
> <https://github.com/tensorflow/tensorflow/pull/9140#discussion_r111070932>
> :
>
> > @@ -0,0 +1,83 @@
> +#!/usr/bin/env bash
>
> That does not seem to work. What is it supposed to do?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/9140#discussion_r111070932>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AT_SbWNxRoaF9ErpVSnhyTeYIMQ3OB_Pks5rvGsmgaJpZM4M6fDc>
> .
>

aselle commented:
It seems I can do
```
#!/bin/bash -u
```
but I don't think I can use env (which gives robustness to bash not installed in /bin)
```
#!/usr/bin/env bash -u
```
doesn't work because the -u doesn't get sent to bash.

drpngx commented:
OK. From the bash script you can just do set -u first thing.

On Wed, Apr 12, 2017 at 11:16 AM, Andrew Selle <notifications@github.com>
wrote:

> It seems I can do
>
> #!/bin/bash -u
>
> but I don't think I can use env (which gives robustness to bash not
> installed in /bin)
>
> #!/usr/bin/env bash -u
>
> doesn't work because the -u doesn't get sent to bash.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/9140#issuecomment-293663003>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AT_Sbb8gY_x9aoc6O8a1ZDWnlTCWvtqvks5rvRTtgaJpZM4M6fDc>
> .
>

aselle committed:
Improve shell script and issue template more (address review)

- Make shell script for stringent by using -u
- Use uppercase GIT_VERSION instead of __git_version__.
- Add OS platform to required information

aselle commented:
@drpngx, your changes are now addressed I believe.

drpngx reviewed (approved):
Sorry I missed this

drpngx commented:
Jenkins, test this please.
